### PR TITLE
gpaddmirrors increase error message clarity

### PR DESCRIPTION
**Pull request by kalensk**
This PR is for 6X _before_ backporting a similar fix to 5x.

Specifically, while on 5X gpaddmirrors can raise the following exception:
`Exception: On host example.com, replication port 27 for segment with dbid 9004 conflicts with a port for segment dbid 26`. Where the port and dbid values got swapped, and the exception could be better worded.

This commit adds tests while making the error messages more clear and concise before backporting to 5X (which will require a few more tests).

### CODE REVIEW

File: gpMgmt/bin/gppylib/operations/test/unit/test_unit_buildMirrorSegments.py [hunk 1]
```diff
@@ -160,5 +168,29 @@ def _createGpArrayWith2Primary2Mirrors(self):
 
         return GpArray([self.master, self.primary0, self.primary1, mirror0, mirror1])
 
+    def test_checkForPortAndDirectoryConflicts__given_the_same_host_checks_ports_differ(self):
+        self.master.hostname = "samehost"
+        self.primary.hostname = "samehost"
+
+        self.master.port = 1111
+        self.primary.port = 1111
+
```
Comment (dkrieger-pvtl) [inline gpMgmt/bin/gppylib/operations/test/unit/test_unit_buildMirrorSegments.py]: It'd be cleaner to set a variable equal to 1111 and use it in the three places that it's referenced in this function.

File: gpMgmt/bin/gppylib/operations/test/unit/test_unit_buildMirrorSegments.py [hunk 2]
```diff
@@ -160,5 +168,29 @@ def _createGpArrayWith2Primary2Mirrors(self):
...
+        with self.assertRaisesRegexp(Exception, r"Segment dbid's 2 and 1 on host samehost cannot have the same port 1111."):
+            self.buildMirrorSegs.checkForPortAndDirectoryConflicts(gpArray)
+
+    def test_checkForPortAndDirectoryConflicts__given_the_same_host_checks_data_directories_differ(self):
+        self.master.hostname = "samehost"
+        self.primary.hostname = "samehost"
+
+        self.master.datadir = "/data"
+        self.primary.datadir = "/data"
```
Comment (dkrieger-pvtl) [inline gpMgmt/bin/gppylib/operations/test/unit/test_unit_buildMirrorSegments.py]: It'd be cleaner to set a variable equal to "/data" and use it in the three places that it's referenced in this function.

**Comment by dkrieger-pvtl** [comment]
LGTM but I think it'd be cleaner to update the code as per the line-specific suggestions.

File: gpMgmt/bin/gppylib/operations/test/unit/test_unit_buildMirrorSegments.py (hunk 1 above)
Comment (kalensk) [reply gpMgmt/bin/gppylib/operations/test/unit/test_unit_buildMirrorSegments.py]: Yes I see what you mean. My preference is to lean on being extra explicit in tests, as well as not needing to always look back at the value of a variable when reading the tests.

I am willing to go the other way if needed, etc.

**Pull request merged**